haas/cli.py

```diff
@@ -179,13 +195,16 @@ def setup_http_client():
 
     This may be extended with other backends in the future.
     """
-    global http_client
```

Comment:
This issue is addressed long back. Now the CLI uses this variable for backward compatibility since client library does not support all calls yet (node_register, switch_register and headnode related calls).